raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -4569,6 +4574,55 @@ public void testObserverFetchWithNoLocalId(boolean withKip853Rpc) throws Excepti
         assertEquals(3, context.log.lastFetchedEpoch());
     }
 
+    @Test
+    public void testExponentialElectionBackoffMs() {
```

Comment:
Did you consider having 3 parametrized tests? On each test invocation you can create a new mocked `Random`. This should allow you to remove `clearInvocations` and be sure that your mock objects are behaving correctly.

Reply:
I chose not to thinking it would overcomplicate the if else conditions based on the number of retries. Let me split this up into two tests actually.